mewarpx/mewarpx/emission.py

```diff
@@ -414,7 +423,7 @@ def inject_particles(self):
             self.record_injectedparticles(
                 self.species,
                 particles_dict['E_total'],
-                particles_dict.get('w', None)
+                particles_dict['w']
```

Comment:
With above change to `record_injectedparticles` the argument order would change.